var/spack/repos/builtin/packages/planck-likelihood/package.py

```diff
@@ -0,0 +1,149 @@
+        md5='7e784819cea65dbc290ea3619420295a',
+        destination='.')
+    resource(
+        name='lensing-ext',
+        url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-lensing-ext.R2.00.tar.gz",
+        md5='091736f73b47a09162050bee27d68399',
+        destination='.',
+        when='+lensing-ext')
+    resource(
+        name='plik-DS',
+        url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-DS.R2.00.tar.gz",
+        md5='76ac04f989025eecab3825aba7e41f36',
+        destination='.',
+        when='+plik-DS')
+    resource(
+        name='plik-HM-ext',
+        url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-HM-ext.R2.00.tar.gz",
+        md5='1c3bd8221f973b7bf7e76647451fd6e5',
+        destination='.',
+        when='+plik-HM-ext')
+    resource(
+        name='plik-unbinned',
+        url="http://irsa.ipac.caltech.edu/data/Planck/release_2/software/COM_Likelihood_Data-extra-plik-unbinned.R2.00.tar.gz",
+        md5='c5869aa6b6581b6863d2a6e0ffd3826c',
+        destination='.',
+        when='+plik-unbinned')
+
+    depends_on('blas')
+    depends_on('cfitsio +shared')
+    depends_on('lapack')
+
+    # Note: Could also install Python bindings
+
+    parallel = False
+
+    def install(self, spec, prefix):
+        # Configure
+
+        # Don't hide build commands
+        filter_file("^\t@", "\t", "Makefile")
+
+        makeflags = [
+            'PREFIX=%s' % prefix,
+            'COLORS=0',
+            'CFITSIOPATH=%s' % spec['cfitsio'].prefix,
+            'CC=cc',
+            'FC=fc',
+            'IFORTLIBPATH=',
+            'IFORTRUNTIME=-lintlc -limf -lsvml -liomp5 -lifportmt -lifcoremt',
+            'GFORTRANLIBPATH=',
+            'GFORTRANRUNTIME=-lgfortran -lgomp',
+            'LAPACKLIBPATH=',
+            'LAPACK=%s' % (spec['lapack'].libs + spec['blas'].libs).ld_flags,
+            'COPENMP=%s' % self.compiler.openmp_flag,
+            'FOPENMP=%s' % self.compiler.openmp_flag,
+        ]
+
+        # Build
+        make(*makeflags)
+
+        # Install
+        make('install', *makeflags)
+        fix_darwin_install_name(prefix.lib)
+        dirs = ['plc_2.0']
+        if '+lensing-ext' in spec:
+            dirs.append('lensing_ext')
+        if '+plik-DS' in spec:
+            dirs.append('plik_DS')
+        if '+plik-HM-ext' in spec:
+            dirs.append('plik_HM_ext')
+        if '+plik-unbinned' in spec:
+            dirs.append('plik_unbinned')
+        for dir in dirs:
+            install_tree(dir, join_path(prefix, 'share', 'clik', dir))
+
+    def setup_dependent_environment(self, module, spec, dep_spec):
+        prefix = self.prefix
+        os.environ['CLIK_PATH'] = prefix
+        os.environ['CLIK_DATA'] = join_path(prefix, 'share', 'clik')
+        os.environ['CLIK_PLUGIN'] = 'rel2015'
```

Comment:
I updated `planck-likelihood`.

I don't think Spack offers the relevant features yet for `cosmomc`. There. I am mostly unsetting environment variables, which is not supported by Spack -- Spack only supports adding or setting variables. Setting an environment variable to the empty string is different from unsetting it.